### ISSUE TITLE
Locomotion scene WALK-ONLY mode stutters

### ISSUE THREAD

**Issue by kevincastejon** [oculus-samples/Unity-StarterSamples#18]
The Locomotion scene offers a WALK mode (AKA no teleport) that stutters a lot, even when enabling Interpolation on the rigidbody. The SimpleCapsuleWithStickMovement.cs is using MovePosition, I tried to switch to setting the velocity directly, which produces smoother move result, but still stutters sometimes regardless of the framerate which is constant 72 FPS.
I ended up testing this samples because I've encountered the exact same issue (stuttering with a rigidbody controlling camera and moved with **velocity = x** with interpolation enabled at constant 72FPS).
I'm really confused to see that the issue is also happening on Oculus officials samples... Is it a Unity physics issue ?
What is the proper way to handle camera rig free move and turn ?

**Comment by kevincastejon** [comment]
It looks like the issue happens on Quest 3, not 2, any thoughts ? May it be my own gear that is faulty?

**Comment by AlexThiv** [comment]
Hi @kevincastejon,

Thanks for bringing this up. We have looked into it and didn't notice as much stutter as you seem to reference to. But there has been some instances:
1) did you notice it on a first play through, after a clean install? I did notice some hiccups when new mesh/textures/materials gets in the view, due to shader loading.
2) I was able to reproduce a bit more stutter when moving back and forth in one of the teleport node and looking on at the floor. But this wasn't really consistent either.

Additionally, this locomotion samples present one of multiple ways that you can implement movement with Unity Physics. We only implement basic movement with the SimpleCapsuleWithStickMovement script, and ideally want developers to optimize movement to their specific needs. This is also a more legacy sample for developers that want a quick understanding on different locomotion using more of the direct apis. You can also look at the Locomotion sample that comes with the **[Interaction SDK package](https://assetstore.unity.com/packages/tools/integration/meta-xr-interaction-sdk-265014)**.

Additional info:
Using the rigidbody velocity is possibly the best way to do this. Alternatively, using a kinematic rigidbody with interpolation and MovePosition works very well, but you have to take care of gravity yourself and disable the HMD Rotate Player checkbox. It is also possible (but not advised) to increase the Physics tick rate in Unity.

Request:
Do you have some more specific repro steps or location where it stutters more?
Do you have a video that demonstrate the stutter?

**Comment by kevincastejon** [comment]
Thanks for answering but unfortunately I had to move on... I'm now using the OpenXR package from Unity which reduces stuttering a lot with its own camera rig.
If I find some time to investigate more, I'll notice you on here, but what I can remember from the latest tests I've done, it was actually not an issue with moving the camera, it was a jittering concerning any movement, event not physics ones. I did make a minimalist scene with just cubes rotating around the player (tested with and without physics), and the stutter was still there.. not big, constant stuttering, but small inconsistent stuttering indeed. It may be my device being faulty since I got the same stutters on Asgarth 2 official Oculus game.
